libsql-wal/src/segment/list.rs

```diff
@@ -0,0 +1,173 @@
+use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
+use std::sync::Arc;
+
+use arc_swap::ArcSwapOption;
+use fst::{map::OpBuilder, Streamer};
+use libsql_sys::ffi::Sqlite3DbHeader;
+use memoffset::offset_of;
+use zerocopy::{AsBytes, FromZeroes};
+
+use crate::error::Result;
+use crate::fs::file::FileExt;
+use crate::segment::{sealed::SealedSegment, Frame};
+
+struct SegmentLink<F> {
+    segment: SealedSegment<F>,
+    next: ArcSwapOption<SegmentLink<F>>,
+}
+
+pub struct SegmentList<F> {
+    head: ArcSwapOption<SegmentLink<F>>,
+    len: AtomicUsize,
+    /// Whether the segment list is already being checkpointed
+    checkpointing: AtomicBool,
+}
```

Comment:
may be its just me, but I find `SegmentLink` and `SegmentList` confusing. I would probably call `SegmentLink` as `SegmentNode` may be, cos most linked list implementations use the term `node`

Reply:
if that can unconfuse you, np